Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt

```diff
@@ -20,313 +21,461 @@ class StubIrMetadataEmitter(
-
-private val StubType.flags: Flags
-    get() = listOfNotNull(
-            Flag.Type.IS_NULLABLE.takeIf { nullable }
-    ).let { flagsOf(*it.toTypedArray()) }
-
-private val TypealiasStub.flags: Flags
-    get() = listOfNotNull(
-            Flag.IS_PUBLIC
-    ).let { flagsOf(*it.toTypedArray()) }
-
-private val FunctionParameterStub.flags: Flags
-    get() = listOfNotNull(
-            Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() }
-    ).let { flagsOf(*it.toTypedArray()) }
-
-private fun AnnotationStub.map(): KmAnnotation {
-    val args = when (this) {
-        AnnotationStub.ObjC.ConsumesReceiver -> TODO()
-        AnnotationStub.ObjC.ReturnsRetained -> TODO()
-        is AnnotationStub.ObjC.Method -> TODO()
-        is AnnotationStub.ObjC.Factory -> TODO()
-        AnnotationStub.ObjC.Consumed -> TODO()
-        is AnnotationStub.ObjC.Constructor -> TODO()
-        is AnnotationStub.ObjC.ExternalClass -> TODO()
-        AnnotationStub.CCall.CString -> mapOf()
-        AnnotationStub.CCall.WCString -> mapOf()
-        is AnnotationStub.CCall.Symbol ->
-            mapOf("id" to KmAnnotationArgument.StringValue(symbolName))
-        is AnnotationStub.CStruct -> TODO()
-        is AnnotationStub.CNaturalStruct -> TODO()
-        is AnnotationStub.CLength -> TODO()
-        is AnnotationStub.Deprecated -> TODO()
+    private val PropertyAccessor.Getter.flags: Flags
+        get() = flagsOfNotNull(
+                Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() },
+                Flag.IS_PUBLIC,
+                Flag.IS_FINAL,
+                Flag.PropertyAccessor.IS_EXTERNAL.takeIf { this is PropertyAccessor.Getter.ExternalGetter }
+        )
+
+    val PropertyStub.setterFlags: Flags
+        get() = if (kind !is PropertyStub.Kind.Var) flagsOf()
+        else kind.setter.flags
+
+    val PropertyAccessor.Setter.flags: Flags
+        get() = flagsOfNotNull(
+                Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() },
+                Flag.IS_PUBLIC,
+                Flag.IS_FINAL,
+                Flag.PropertyAccessor.IS_EXTERNAL.takeIf { this is PropertyAccessor.Setter.ExternalSetter }
+        )
+
+    val StubType.flags: Flags
+        get() = flagsOfNotNull(
+                Flag.Type.IS_NULLABLE.takeIf { nullable }
+        )
+
+    val TypealiasStub.flags: Flags
+        get() = flagsOfNotNull(
+                Flag.IS_PUBLIC
+        )
+
+    val FunctionParameterStub.flags: Flags
+        get() = flagsOfNotNull(
+                Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() }
+        )
+
+    val ClassStub.flags: Flags
+        get() = flagsOfNotNull(
+                Flag.HAS_ANNOTATIONS.takeIf { annotations.isNotEmpty() },
+                Flag.IS_PUBLIC,
+                Flag.IS_OPEN.takeIf { this is ClassStub.Simple && this.modality == ClassStubModality.OPEN },
+                Flag.IS_FINAL.takeIf { this is ClassStub.Simple && this.modality == ClassStubModality.NONE },
```

Comment:
Why not reuse `modality.flags` then?

Reply:
It will require additional refactoring that can be performed separately.